snuba/query/project_extension.py

```diff
@@ -112,9 +112,43 @@ def do_post_processing(
                 if len(exclude_group_ids) > max_group_ids_exclude:
                     query.set_final(True)
                 else:
-                    query.add_conditions(
-                        [(["assumeNotNull", ["group_id"]], "NOT IN", exclude_group_ids)]
-                    )
+                    # Check if there is a more selective conditions around group_id/issue
+                    # in the query conditions. Privilege the positive conditions instead of
+                    # the negative ones.
+                    strict_issue_conditions = False
+                    replaced_conditions: List[Condition] = []
+                    for c in query.get_conditions():
+                        new_condition = c
+                        if (
+                            util.is_condition(c)
+                            and c[0] in ["group_id", "issue"]
+                            and c[1] == "IN"
+                            and isinstance(c[2], list)
+                            and all(isinstance(group, int) for group in c[2])
+                        ):
+                            group_ids_with_exclusions = [
+                                group
+                                for group in c[2]
+                                if group not in exclude_group_ids
+                            ]
+                            if group_ids_with_exclusions:
```

Comment:
If this is an `IN` query that contains 0 items since they were all in the exclusion set, isn't that still significant here?